src/f32ext/ln.rs

```diff
@@ -0,0 +1,113 @@
+use super::abs;
+use super::utils;
+use super::utils::FloatComponents;
+/// Floating natural log
+use core::f32;
+use core::u32;
+
+//excessive precision doesn't matter that much.
+#[allow(clippy::excessive_precision)]
+pub(super) fn ln_1to2_series_approximation(x: f32) -> f32 {
+    // idea from https://stackoverflow.com/a/44232045/
+    // modified to not be restricted to int range and only values of x above 1.0.
+    // and got rid of most of the slow conversions,
+    // should work for all positive values of x.
+    // log base 2(E) == 1/ln(2)
+    //let recip_ln2: f32 = f32::consts::LOG2_E;
```

Comment:
Why is this line here but commented out?

Reply:
I had commented it out when compiler warnings complained it wasn't used and I forgot to get rid of it, I believe a similar line exists in exp, and I preemptively put 1/ln(2) in that file thinking I'd need it, and didn't.